data/transforms/lakh.py

```diff
@@ -0,0 +1,71 @@
+# from random import randint # TODO should the transform sample the midi sequence at random?
+from miditok import REMI, TokenizerConfig, TokSequence
+from random import randint
+from typing import List
+
+TOKENIZER_PARAMS = {
+    "pitch_range": (0, 127),
+    "vocab_size": 130,  # Total vocab size: 130 for pitches, note-off, rest + 512 for drum patterns
+    "max_bar_length": 16,  # Each bar has 16 events (16th notes)
+    "num_bars": 16,  # Total number of bars for hierarchical model
+    "hierarchical": True,  # Use hierarchical modeling
+    "bar_token_count": 16,  # Each subsequence corresponds to a single bar
+    "beat_resolution": 4,  # 16th note intervals (4 intervals per beat in 4/4 time)
+    "note_on_count": 128,  # 128 note-on tokens
+    # "use_rests": True,
+    # "drum_pattern_count": 512  # 512 categorical tokens for drum patterns
+}
+
+
+class LakhTransform(object):
```

Comment:
Ok so here's the thing - the idea behind those transforms is to make them as simple and as generic as possible (basically we want them to satisfy SRP). That way we can later reuse and combine them. They are supposed to resemble the pytorch (torchvision) tranforms. In pytorch, you can for example define transform pipeline that looks like this:
```
transform_pipeline = Compose(
    [
        ToPILImage(),                                   # tranform numpy array to PIL Image object
        RandomHorizontalFlip(),                         # randomly flip the image horizontally with probability 0.5
        RandomCrop(32, padding=4),                      # randomly crop the image
        Grayscale(),                                    # what the name says
        ToTensor(),                                     # turn PIL Image object into a pytorch Tensor
        Normalize([0.5, 0.5, 0.5], [0.5, 0.5, 0.5])     # normalize the output
    ]
)
# The Compose basically does this when you call it:
# def __call__(self, image):
#       out = image
#       for transform in self.transform_list:
#           out = tranform(out)
#       return out
``` 
Those transforms do some very generic things (change image into tensor, flip the image etc.) We probably want our transforms to be similarly generic.

Ok, now from what I saw this transform does 4 things:
* loads the file and tokenizes it with miditok
* filters out the drum tracks
* selects track with the biggest amount of non-empty bars
* reduces the track to randomly chosen 16 bars

In other words, we probably want to refactor this into those separate parts (also see other comments about selecting the track). **If we have time for that** that is. Otherwise, let's just leave a TODO or an issue. This is a **low priority** I think.

Reply:
Transform is now simplified after refactor